### ISSUE TITLE
_Memory_Block: units' weights are need to be optimized by Adam optimizer ?

### ISSUE THREAD

**Issue by YChienHung** [hhb072/MUSS#1]
In your paper, the weights of the units serve as the memory bank data, and they are updated using EMA. However, I noticed that they can also be updated by the optimizer in your code. 
Which is the correct approach?
```python
class _Memory_Block(nn.Module):        
    def __init__(self, hdim, kdim, max_T=1., min_T=0.03, decay_alpha=0.99998, moving_average_rate=0.999):
        super().__init__()
        
        self.c = hdim
        self.k = kdim
        self.moving_average_rate = moving_average_rate
                
---> self.units = nn.Embedding(kdim, hdim)
        
        self.max_T = max_T
        self.min_T = min_T
        self.decay_alpha = decay_alpha
        self.register_buffer('cur_T', torch.tensor(max_T))
                
```

**Comment by hhb072** [comment]
Hello,
The parameters won't be optimized by the Adam optimizer because we use m = self.units.weight.data in the forward function rather than self.units.weight directly. By accessing .data, we bypass PyTorch's automatic differentiation mechanism (autograd), which means gradients aren't computed for these parameters. As a result, the optimizer doesn't update them during training.

**Comment by YChienHung** [comment]
Thanks for your enlightening answer !
